integration_test/common/common_test.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build integration_test
+
+package common
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path"
+	"strings"
+	"testing"
+
+	"github.com/go-playground/validator/v10"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	//relative path to testdata folder
+	testdataDir   = "testdata"
+	inputYamlName = "input.yaml"
+)
+
+type testCase struct {
+	dirName        string
+	expectedErrors map[fieldError]struct{}
+}
+
+type fieldError struct {
+	field string
+	tag   string
+}
+
+func getInputYamlBytes(t *testing.T, dirName string) []byte {
+	yamlFilePath := path.Join(testdataDir, dirName, inputYamlName)
+	contents, err := ioutil.ReadFile(yamlFilePath)
+	if err != nil {
+		t.Fatal("could not read dirName: " + yamlFilePath)
+	}
+	escapedYamlStr := strings.ReplaceAll(string(contents), "\r\n", "\n")
+
+	return []byte(escapedYamlStr)
+}
+
+func UnmarshallAndValidate(t *testing.T, bytes []byte, i interface{}) error {
+	v := validator.New()
+	err := yaml.UnmarshalStrict(bytes, i)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return v.Struct(i)
+}
+
+func TestAll(t *testing.T) {
+	table := []testCase{
```

Comment:
Have we considered using a `golden_error` file in the same directory? Similar to https://github.com/GoogleCloudPlatform/ops-agent/tree/master/confgenerator/testdata/invalid/linux/logging-processor_exclude_logs_invalid_field

The potential benefit is to skip the step to update this list. "invalid case" and "errors" will be easily found together. And if users forgets to update this list, the test will still executes. With the current approach, if someone adds an invalid `input.yaml` but forget or is not aware that they need to update it here, the tests might be skipped by accident.

There might be downside I'm not seeing though.